docs/ldml/tr35-keyboards.md

```diff
@@ -1348,206 +1342,615 @@ This example shows some of the mappings for a French keyboard layout:
+
+**Implementation Notes**
+
+Ideally, markers are implemented entirely out-of-band from the normal text stream. However, implementations _may_ choose to map each marker to a [Unicode private-use character](https://www.unicode.org/glossary/#private_use_character) for use only within the implementation’s processing and temporary storage in the input context.
+
+For example, the first marker encountered could be represented as U+E000, the second by U+E001 and so on.  If a regex processing engine were used, then those PUA characters could be processed through the existing regex processing engine.  `[^\uE000-\uE009]` could be used as an expression to match a character that is not a marker, and `[Ee]\u{E000}` could match `E` or `e` followed by the first marker.
+
+Such implementations must take care to remove all such markers (see prior section) from the resultant text. As well, implementations must take care to avoid conflicts if applications themselves are using PUA characters, such as is often done with not-yet-encoded scripts or characters.
+
+* * *
+
+### <a name="Element_transformGroup" href="#Element_transformGroup">Element: transformGroup</a>
+
+> <small>
+>
+> Parents: [transforms](#Element_transforms)
+>
+> Children: [import](#Element_import), [reorder](#Element_reorder), [_special_](tr35.md#special), [transform](#Element_transform)
+>
+> Occurrence: optional, multiple
+> </small>
+
+A `transformGroup` represents a set of transform elements or reorder elements.
+
+Each `transformGroup` is processed entirely before proceeding to the next one.
+
+
+Each `transformGroup` element, after imports are processed, must have either [reorder](#Element_reorder) elements or [transform](#Element_transform) elements, but not both. The `<transformGroup>` element may not be empty.
+
+**Examples**
+
+#### Example: `transformGroup` with only `import` elements
+
+This is a `transformGroup` that consists only of [`import`](#element-import) elements.  With this form, only `transform` OR `reorder` elements may be imported by all import statements, but not both.
+
+```xml
+<transformGroup>
+    <import path="..."/>
+    <!-- other <import> elements -->
+</transformGroup>
+```
+
+
+#### Example: `transformGroup` with `transform` elements
+
+This is a `transformGroup` that consists of one or more [`transform`](#element-transform) elements, prefaced by one or more `import` elements. See the discussion of those elements for details.
+
+```xml
+<transformGroup>
+    <import path="..."/> <!-- zero or more optional import elements-->
+    <transform />
+    <!-- other <transform/> elements -->
+</transformGroup>
+```
+
+
+#### Example: `transformGroup` with `reorder` elements
+
+This is a `transformGroup` that consists of one or more [`reorder`](#element-reorder) elements, prefaced by one or more `import` elements. See the discussion of those elements for details.
```

Comment:
```suggestion
This is a `transformGroup` that consists of one or more [`reorder`](#element-reorder) elements, optionally prefaced by one or more `import` elements that import `reorder` elements. See the discussion of those elements for details.

`import` elements in this group type may not import `transform` elements.
```

With this change and its corresponding change for `transformGroup` with `transform` elements, I don't see a reason for "`transformGroup` with only `import` elements" to exist, except _maybe_ as a footnote to these two.

Reply:
correct, it should just go